yesod-core/Yesod/Core/FormHelpers.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Yesod.Core.FormHelpers
+    (
+      HttpMethod(..),
+      FormStyle,
+      httpLink,
+      buttonLink,
+      link
+    ) where
+
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.Trans.Control (MonadBaseControl (..))
+import Yesod.Routes.Class
+import Control.Monad.Trans.Resource (MonadResource (..), MonadThrow (..))
+import Yesod.Core.Widget
+import Yesod.Core.Handler
+import Text.Blaze (ToMarkup)
+import Data.String (IsString)
+import Text.Lucius
+
+data HttpMethod = HttpGET | HttpPOST | HttpPATCH | HttpUPDATE | HttpDELETE
+data FormStyle = ButtonForm | LinkForm
+
+getMethodType :: (ToMarkup a, IsString a) => HttpMethod -> a
+getMethodType method =
+  case method of
+    HttpGET -> "get"
+    HttpPOST -> "post"
+    HttpPATCH -> "patch"
+    HttpUPDATE -> "patch"
+    HttpDELETE -> "delete"
+
+isStandardHttpMethod :: HttpMethod -> Bool
+isStandardHttpMethod httpMethod =
+  case httpMethod of
+    HttpGET -> True
+    HttpPOST -> True
+    HttpPATCH -> False
+    HttpUPDATE -> False
+    HttpDELETE -> False
+
+httpLink :: (MonadIO m, MonadBaseControl IO m, MonadThrow m, ToMarkup a) => FormStyle -> HttpMethod -> Route site -> a -> WidgetT site m ()
+httpLink formType methodType path text = do
+  request <- getRequest
+  let method = (getMethodType methodType) :: String
+      standardHttpMethod = isStandardHttpMethod methodType
+  toWidget
+    [whamlet|
+      <form action=@{path} method=#{method}>
```

Comment:
Ahhh, I see what you're saying. I did that because my understanding is that setting the method= to an invalid method defaults to GET and I thought it would make it a bit more clear as to what is happening in the html. I will default it to POST.